marklogic-data-hub-central/ui/src/components/entities/mapping/mapping-step-detail/mapping-step-detail.tsx

```diff
@@ -478,19 +475,22 @@ const MappingStepDetail: React.FC = () => {
       let entEntityTempData: any = [];
       let nestedEntityProps = extractNestedEntityData(entProps, entEntityTempData);
       setEntityTypeProperties([...nestedEntityProps]);
-      setTgtEntityReferences({...tgtRefs});
       setTargetRelatedMappings(resp.data[0]["relatedEntityMappings"]);
       if (resp.data[0]["relatedEntityMappings"] && resp.data[0]["relatedEntityMappings"].length > 0) {
         resp.data.forEach(entityObject => {
           let relatedEntityName = entityObject["entityType"];
           if (relatedEntityName !== curationOptions.activeStep.entityName && entityObject["entityModel"].definitions && entityObject.mappingTitle && entityObject["entityMappingId"]) {
             let relatedEntProps = entityObject["entityModel"].definitions[relatedEntityName].properties;
             let relatedEntityTempData: any =[];
+            let contextKey = EntityTableKeyIndex + 1;
+            EntityTableKeyIndex++;
             let relatedEntityProps = extractNestedEntityData(relatedEntProps, relatedEntityTempData);
-            setRelatedEntityTypeProperties(prevState => ([...prevState, {entityType: entityObject.entityType, entityLabel: entityObject.mappingTitle, entityMappingId: entityObject.entityMappingId, relatedEntityMappings: entityObject.relatedEntityMappings, entityProps: relatedEntityProps}]));
+            relatedEntityProps.unshift({key: contextKey, name: "Context", type: "", isProperty: false, filterName: "Context", filterMatch: false}); //add Context field to front of properties
+            setRelatedEntityTypeProperties(prevState => ([...prevState, {entityType: entityObject.entityType, entityModel: entityObject.entityModel, entityLabel: entityObject.mappingTitle, entityMappingId: entityObject.entityMappingId, relatedEntityMappings: entityObject.relatedEntityMappings, entityProps: relatedEntityProps}]));
```

Comment:
This line sometimes cause duplication of the entity objects in the relatedEntityTypeProperties array. 
<img width="1102" alt="Screen Shot 2021-04-27 at 11 44 23 AM" src="https://user-images.githubusercontent.com/44847151/116305556-ca6c7c00-a758-11eb-8685-b2279f732a0a.png">

Reply:
Thanks @timur-isangulov for the catch and assistance. Pushed a fix.